testservices/identityservice/v3userpass.go

```diff
@@ -203,6 +205,11 @@ func (u *V3UserPass) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		u.ReturnFailure(w, http.StatusInternalServerError, err.Error())
 		return
 	}
+	if req.Auth.Scope.Project.ID != "" {
```

Comment:
This block is wrong - you are setting res.Project twice and incorrectly calling addTenant
You will need to tweak addTenant() and also look to pass in the project name